apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/grpc/ToProtoValueCodecs.java

```diff
@@ -832,6 +832,12 @@ public QueryOuterClass.Value protoValueFromStrictlyTyped(Object mapValue) {
           List<QueryOuterClass.Value> elements = new ArrayList<>();
           for (Object entry : (List<?>) mapValue) {
             Map<?, ?> entryNode = (Map<?, ?>) entry;
+            if (entryNode.size() != 2) {
+              throw new IllegalArgumentException("Map entry must have exactly 2 elements");
+            }
+            if (!entryNode.containsKey("key") || !entryNode.containsKey("value")) {
+              throw new IllegalArgumentException("Map entry missing key or value");
```

Comment:
Maybe just separate checks so you can specifically say if it is missing key or missing value

Reply:
Sure, I split the check now.